src/timeline.ts

```diff
@@ -193,87 +223,105 @@ class Timeline implements IAppView {
       svgtimeline.attr('width', widthTimelineDiv);
       d3.select('line').attr('x2', widthTimelineDiv);
 
-      //Updating scale for circle position
-      xScaleTime.range([20, widthTimelineDiv-20]);
+      // Updating scale for circle position
+      xScaleTime.range([20, widthTimelineDiv - 20]);
 
       svgtimeline.selectAll('circle')
-        .attr('cx', (d:any, i ) => {
-          if(d.time) {
-            return xScaleTime(moment(d.time).diff(moment(items[0].time),'days'));
+        .attr('cx', (d: any, i) => {
+          if (d.time) {
+            return xScaleTime(moment(d.time).diff(moment(items[0].time), 'days'));
           } else {
             return i * scaleCircles(widthTimelineDiv);
           }
         });
 
-      //Update bars
+      // Update bars
       svgtimeline.selectAll('g').remove();
 
-      if(widthTimelineDiv <= 800) {
+      if (widthTimelineDiv <= 800) {
         if (rectWidth >= 5) {
           svgtimeline.selectAll('g').remove();
         } else {
-          rectWidth = rectWidth-1;
+          rectWidth = rectWidth - 1;
           generateBars(rectWidth);
         }
       } else {
         rectWidth = 15;
         generateBars(rectWidth);
       }
-    };
+    }
 
-    //creating 2D Ratio bars
+
+
+    // helper variable for on click event
+    let open2dHistogram = null;
+
+    // creating 2D Ratio bars
     function generateBars(width) {
```

Comment:
Again, avoid nested functions. Flatten the function structure and use `this.generateBars()` to reference the class method.